File: DSA/Manav-Rajput/Day37_question2.java
class Solution {
    public boolean kPangram(String str, int k) {
        boolean[] present = new boolean[26];
        for (char ch : str.toCharArray()) {
            if (ch >= 'a' && ch <= 'z') {
                present[ch - 'a'] = true;
            }
        }
        
        int additionalNeeded = 0;
        for (boolean isPresent : present) {
            if (!isPresent) {
                additionalNeeded++;
            }
        }
      
        return additionalNeeded <= k;
    }
}

File: DSA/Arshdeep-Kaur/Day 37/Day37_Question2.cpp
class Solution {
public:
    bool kPangram(string str, int k) {
        int count[26] = {0};
        int n = str.length();
        
        for (int i = 0; i < n; i++) {
            if (str[i] >= 'a' && str[i] <= 'z') {
                count[str[i] - 'a']++;
            } else if (str[i] >= 'A' && str[i] <= 'Z') {
                count[str[i] - 'A']++;
            }
        }
        
        int missingLetters = 0;
        for (int i = 0; i < 26; i++) {
            if (count[i] == 0) {
                missingLetters++;
            }
        }
        
        return missingLetters <= k;
    }
};
